File: 013_Roman_to_Integer/013_3.java
class Solution{
    public int romanToInt(String s) {
        int total = 0;
        char prev = 'A';
        for (char c : s.toCharArray()) {
            switch(c) {
                case 'M': total += prev == 'C' ? 800 : 1000;
                          break;
                case 'D': total += prev == 'C' ? 300 : 500;
                          break;
                case 'C': total += prev == 'X' ? 80 : 100;
                          break;
                case 'L': total += prev == 'X' ? 30 : 50;
                          break;
                case 'X': total += prev == 'I' ? 8 : 10;
                          break;
                case 'V': total += prev == 'I' ? 3 : 5;
                          break;
                case 'I': total += 1;
            }
            prev = c;
        }
        return total;
    }
}

File: 013_Roman_to_Integer/013_3.py
class Solution:
    def romanToInt(self, s):
        maps = {'I':1, 'V':5, 'X':10, 'L':50, 
                'C':100, 'D':500, 'M':1000}

        total = 0
        prev = 0
        for char in s:
            curr = maps.get(char)
            total += (curr - 2 * prev) if (curr > prev) else curr
            prev = curr
        return total